File: python-examples/quick-recipes/api/download-file.py
from typing import TypedDict

from intuned_browser import download_file
from playwright.async_api import Page


class Params(TypedDict):
    pass


async def automation(page: Page, params: Params | None = None, **_kwargs):
    # Navigate to the sample downloads page
    await page.goto("https://sandbox.intuned.dev/pdfs")

    # Locate the download button for the first row
    download_locator = page.locator("xpath=//tbody/tr[1]//*[name()='svg']")

    # Trigger download and wait for the file
    # https://docs.intunedhq.com/automation-sdks/intuned-sdk/python/helpers/functions/download_file
    downloaded_file = await download_file(
        page=page,
        trigger=download_locator,
        timeout_s=15,
    )

    # Return the suggested filename for reference
    file_name = downloaded_file.suggested_filename
    print(f"Downloaded file: {file_name}")
    return {
        "file_name": file_name,
    }

File: typescript-examples/quick-recipes/api/download-file.ts
import { BrowserContext, Page } from "playwright";
import { downloadFile } from "@intuned/browser";

interface Params {
  // No params needed
}

export default async function handler(
  params: Params,
  page: Page,
  context: BrowserContext
) {
  // Navigate to the sample downloads page
  await page.goto("https://sandbox.intuned.dev/pdfs");

  // Locate the download button for the first row
  const downloadLocator = page.locator("xpath=//tbody/tr[1]//*[name()='svg']");

  // Trigger download and wait for the file
  // https://docs.intunedhq.com/automation-sdks/intuned-sdk/typescript/helpers/functions/downloadFile
  const downloadedFile = await downloadFile({
    page,
    trigger: downloadLocator,
    timeoutInMs: 15000,
  });

  // Return the suggested filename for reference
  const fileName = downloadedFile.suggestedFilename();
  console.log(`Downloaded file: ${fileName}`);
  return fileName;
}
